command/agent/http.go

```diff
@@ -65,8 +65,8 @@ func NewHTTPServer(agent *Agent, config *Config) (*HTTPServer, error) {
 	// If TLS is enabled, wrap the listener with a TLS listener
 	if config.TLSConfig.EnableHTTP {
 		tlsConf := &tlsutil.Config{
-			VerifyIncoming:       false,
-			VerifyOutgoing:       true,
+			VerifyIncoming:       config.TLSConfig.VerifyHTTPSClient,
+			VerifyOutgoing:       config.TLSConfig.VerifyHTTPSClient,
```

Comment:
This should still be hard-coded as `true`. If `TLSConfig.EnableHTTP` is set we don't want to even allow skipping server certificate verification.